dialogue-core/src/main/java/com/palantir/dialogue/core/ConcurrencyLimitedChannel.java

```diff
@@ -97,13 +98,9 @@ static CautiousIncreaseAggressiveDecreaseConcurrencyLimiter createLimiter() {
         if (maybePermit.isPresent()) {
             CautiousIncreaseAggressiveDecreaseConcurrencyLimiter.Permit permit = maybePermit.get();
             logPermitAcquired();
-            Optional<ListenableFuture<Response>> result = delegate.maybeExecute(endpoint, request);
-            if (result.isPresent()) {
-                DialogueFutures.addDirectCallback(result.get(), permit);
-            } else {
-                permit.ignore();
-            }
```

Comment:
this branching was also pointless - we'd literally never end up in this `else` case because the adapter always returned Optional.of